### PR TITLE
Add test for large BLOB with parameter

### PR DESCRIPTION
**Pull request by illuusio**
Add test for large binary BLOB inserting with prepared statement and large Gzip binary as parameter.

### CODE REVIEW

**Comment by Grinnz** [comment]
As noted in #179 the correct application behavior is to bind the parameter as binary when inserting it.

**Comment by illuusio** [comment]
Ok now using correct `bind_param` in test and it fixes things :dancers:. Rebased also wrong commit away

**Comment by pali** [comment]
Hello! I would like to ask what you motivated to write this test without any code change? Have you triggered some issue in the driver code? Or is there some known bug related to LONGBLOB type?

About test itself, I do not like two things: Usage of rand() and dropping table at the beginning of the test. Test should be deterministic, not random because it would be harder to debug it if it fails only sometimes. And instead of dropping table at the beginning of the test you can use temporary table which is visible only for the current $dbh connection, and is automatically dropped after closing it.

Anyway, repository now contains new Github Actions CI which run tests from the every pull request against lot of different client and server versions. So after rebasing your PR on top of the master branch, github runs these new tests automatically.

**Comment by pali** [comment]
Btw, when you are adding usage of a new module in test as hard dependency then it is needed to specify that module in TEST_REQUIRES section in Makefile.PL.

**Comment by illuusio** [comment]
Should this be closed altogether as this was just me who was thinking that it could be nice to have this kind of test. As I faced this when testing Perl CHI DBI driver and it didn't work with MariaDB but it worked just fine with MySQL, Postgres and SQLite. As said this just RTFM like people like who don't read Pod/Man pages.

**Comment by pali** [comment]
I'm not against such test. Just fix issues in the test which I mentioned above.

**Comment by illuusio** [comment]
Ok i've updated this test but have to solve `max_allowed_packet` problem on older systems

**Comment by pali** [comment]
@illuusio could you fix test to pass CI?

**Comment by illuusio** [comment]
Sorry this dropped out of my scope for other things but let's see if 512 KB is small enough to make old MySQL and MariaDB happy. I've also rebased branch.

File: t/40blobslarge.t [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    for (my $i = 0;  $i < 8;  $i++) {
+        if (defined($blob)  &&  length($blob) > $i) {
+            $b = substr($blob, $i*32);
+        }
+        else {
+            $b = "";
+        }
+        note sprintf("%08lx %s\n", $i*32, unpack("H64", $b));
+    }
+}
```
Comment (choroba) [inline t/40blobslarge.t:22]: This sub is never used.

Comment (illuusio) [reply t/40blobslarge.t]: Thank you for pointing out. I've removed it and reworded comments little bit

**Comment by pali** [comment]
@choroba any other comments?

**Comment by choroba** [comment]
No more comments 🙂

**Review by pali — APPROVED** [review]

**Pull request merged**